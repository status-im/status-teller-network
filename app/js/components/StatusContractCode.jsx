import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Row, Col, Card } from 'reactstrap';
import { Link } from "react-router-dom";
import { withNamespaces } from 'react-i18next';
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faQrcode } from "@fortawesome/free-solid-svg-icons";

class StatusContractCode extends Component {
  render() {
    const t = this.props.t;
    return (
      <Row className="mt-4">
        <Col xs="12">
          <span className="font-weight-bold h5">{t('statusContractCode.title')}</span>
          <Link to="edit-status-contract-code" className="float-right">{t('statusContractCode.edit')}</Link>
        </Col>
        <Col xs="10" className="pr-0">
          <Card body className="rounded-0">
            <span>{this.props.value}</span>
          </Card>
        </Col>
        <Col xs="2" className="pl-0">
          <Card body className="text-center rounded-0">
            <span><FontAwesomeIcon size="lg" icon={faQrcode}/></span>
          </Card>
        </Col>
      </Row>
    );
  }
}

StatusContractCode.propTypes = {
  t: PropTypes.func,
  value: PropTypes.string
};

export default withNamespaces()(StatusContractCode);
