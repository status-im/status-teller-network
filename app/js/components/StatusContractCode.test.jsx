import React from "react";
import { shallow } from "enzyme";

import StatusContractCode from "./StatusContractCode";

describe('StatusContractCodeades', () => {
  it('should render correctly', () => {
    const component = shallow(<StatusContractCode value={"0xkajshdhkjashd"} />);
  
    expect(component).toMatchSnapshot();
  });
});
