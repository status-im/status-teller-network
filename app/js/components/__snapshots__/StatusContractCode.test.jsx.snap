// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusContractCodeades should render correctly 1`] = `
<Row
  className="mt-4"
  tag="div"
>
  <Col
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
    xs="12"
  >
    <span
      className="font-weight-bold h5"
    />
    <Link
      className="float-right"
      replace={false}
      to="edit-status-contract-code"
    />
  </Col>
  <Col
    className="pr-0"
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
    xs="10"
  >
    <Card
      body={true}
      className="rounded-0"
      tag="div"
    >
      <span>
        0xkajshdhkjashd
      </span>
    </Card>
  </Col>
  <Col
    className="pl-0"
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
    xs="2"
  >
    <Card
      body={true}
      className="text-center rounded-0"
      tag="div"
    >
      <span>
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                448,
                512,
                Array [],
                "f029",
                "M0 224h192V32H0v192zM64 96h64v64H64V96zm192-64v192h192V32H256zm128 128h-64V96h64v64zM0 480h192V288H0v192zm64-128h64v64H64v-64zm352-64h32v128h-96v-32h-32v96h-64V288h96v32h64v-32zm0 160h32v32h-32v-32zm-64 0h32v32h-32v-32z",
              ],
              "iconName": "qrcode",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size="lg"
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
      </span>
    </Card>
  </Col>
</Row>
`;
